recipes/caf/all/conanfile.py

```diff
@@ -0,0 +1,140 @@
+    def _is_static(self):
+        return 'shared' not in self.options.values.keys() or not self.options.shared
+
+    @property
+    def _has_openssl(self):
+        return 'openssl' in self.options.values.keys() and self.options.openssl
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            if self.settings.arch == "x86":
+                del self.options.openssl
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("actor-framework-" + self.version, self._source_subfolder)
+
+    def requirements(self):
+        if self._has_openssl:
+            self.requires("openssl/1.1.1j")
+
+    def configure(self):
+        if not self._is_static and self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Shared libraries are not supported on Windows")
+        compiler_version = Version(self.settings.compiler.version.value)
+        # Will remove after https://github.com/actor-framework/actor-framework/issues/1226 fixed
+        if self.version != "0.17.6" and self.settings.compiler == "clang" and compiler_version == "5":
+            raise ConanInvalidConfiguration("caf 0.18.0+ not supported on clang 5")
+        if self.version != "0.17.6" and \
+            (self.settings.compiler == "gcc" and compiler_version < "7") or \
+                (self.settings.compiler == "clang" and compiler_version < "5") or \
+                (self.settings.compiler == "apple-clang" and compiler_version < "10") or \
+                (self.settings.compiler == "Visual Studio" and compiler_version < "16"):
+            raise ConanInvalidConfiguration("caf 0.18.0+ requires a C++17 compiler")
+        if self.settings.compiler == "gcc":
+            if compiler_version < "4.8":
+                raise ConanInvalidConfiguration("g++ >= 4.8 is required, yours is %s" % self.settings.compiler.version)
+        elif self.settings.compiler == "clang" and compiler_version < "4.0":
+            raise ConanInvalidConfiguration("clang >= 4.0 is required, yours is %s" % self.settings.compiler.version)
+        elif self.settings.compiler == "apple-clang" and compiler_version < "9.0":
+            raise ConanInvalidConfiguration("clang >= 9.0 is required, yours is %s" % self.settings.compiler.version)
+        elif self.settings.compiler == "apple-clang" and compiler_version > "10.0" and \
+                self.settings.arch == 'x86':
+            raise ConanInvalidConfiguration("clang >= 11.0 does not support x86")
+        elif self.settings.compiler == "Visual Studio" and compiler_version < "15":
+            raise ConanInvalidConfiguration("Visual Studio >= 15 is required, yours is %s" % self.settings.compiler.version)
+
+    def _cmake_configure(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            if self.version == "0.17.6":
+                self._cmake.definitions["CMAKE_CXX_STANDARD"] = "11"
```

Comment:
```suggestion
```

cxx standard cannot be forced here, otherwise, when passing a different compiler.cppstd value won't work

Reply:
you can pass `self._cmake.definitions["CMAKE_CXX_STANDARD"] = self.compiler.get_safe("cppstd", "11")` if caf requires C++11.

Reply:
```suggestion
                self._cmake.definitions["CMAKE_CXX_STANDARD"] = self.compiler.get_safe("cppstd", "11")
```